app/controllers/carts_controller.rb

```diff
@@ -9,15 +9,18 @@ def create
   end
 
   def show
-    if current_user.recipient?
+    if current_user && current_user.recipient?
       @needs = @cart.create_cart_needs || {}
-      require "pry"; binding.pry
     else
-      recipients = User.find(@cart.contents["donor"].keys)
-      require "pry"; binding.pry
-      @recipients_donations = recipients.inject({}) do |hash, recipient|
-        hash[recipient] = recipient.donations
-        hash
+      if @cart.contents["donor"]
```

Comment:
we'll do this tomorrow....but we need to remove this logic out of the controller.